src/index.ts

```diff
@@ -48,3 +52,23 @@ export function* select<Args extends any[], R>(
 export function* cancelled(): SagaGenerator<boolean> {
   return yield rawCancelled();
 }
+
+export function* put<T extends Action>(action: T): SagaGenerator<T> {
+  return yield rawPut(action);
+}
+
+export function race<T extends object>(
```

Comment:
This one is a bit tricky, it works well when inputs are wrappers from `typed-redux-saga` but does not infer correct types when effects are used from `redux-saga`.

E.g. using wrappers from typed saga
```ts
import { race, call, take } from 'typed-redux-saga';

const fetchData = () => ({ foo: true });

const { data, cancel } = yield* race({
  data: call(fetchData),
  cancel: take('cancel')
});

// race results are inferred correctly:
// data: { foo: string } | undefined
// cancel: Action<any> | undefined
``` 
E.g. using effects from redux-saga
```ts
import { call, take } from 'redux-saga';
import { race } from 'typed-redux-saga';

const fetchData = () => ({ foo: true });

const { data, cancel } = yield* race({
  data: call(fetchData),
  cancel: take('cancel')
});

// race results are incorrectly inferred:
// data: SimpleEffect<"CALL", CallEffectDescriptor<{ foo: boolean; }>> | undefined
// cancel: SimpleEffect<"TAKE", TakeEffectDescriptor> | undefined
```

In the latter example, I guess it's still possible to get back correct type info with some complex typing magic.